app/controllers/api/v1/transactions_controller.rb

```diff
@@ -43,19 +45,31 @@ def create
         end
       end
 
+      rescue_from CentsMustBeInteger do
+        render json: { errors: [{ detail: "Cents must be an integer" }] }, status: :unprocessable_entity
+      end
+
       private
 
       def transaction_params
-        t = params.require(:transaction).permit(:debtor, :creditor, :message, :euros, :cents, :id_at_client)
+        t = params.require(:transaction).permit(:debtor, :creditor, :message, :cents, :id_at_client)
+
+        validate_cents(t[:cents])
 
         {
           debtor: t[:debtor] ? User.find_by(name: t[:debtor]) : User.zeus,
           creditor: t[:creditor] ? User.find_by(name: t[:creditor]) : User.zeus,
           issuer: current_user_or_client,
-          amount: ((BigDecimal(t[:euros]&.to_s || "0") * 100) + t[:cents].to_i).to_i,
+          amount: t[:cents].to_i,
           message: t[:message],
         }.merge(current_client ? { id_at_client: t[:id_at_client] } : {})
       end
+
+      def validate_cents(cents)
+        return if cents.nil?
+
+        raise CentsMustBeInteger unless cents.to_s.match?(/^-?\d+$/)
```

Comment:
How about validating this in the model using [`validates_numericality_of`](https://api.rubyonrails.org/classes/ActiveModel/Validations/HelperMethods.html#method-i-validates_numericality_of) with `:only_integer`?

Reply:
Done! 😄